### PR TITLE
Changed Summarizers for the Visualizer.

### PR DESCRIPTION
**Pull request by ClintonCao**
I've adapted the summarizers for the Visualizer. The visualizer needed the following things:

- Number of classes in a package.
- Number of CheckStyle Warnings per package / component.
- Number of PMD warnings per package / component.
- Number of FindBugs Warnings per package / component.

Though this is finished, I suggest we merge #23 first before we merge this pull request. The reason is because I updated the folder structure in that pull request. It is less of a work to fix the conflicts in this PR.

If there are any questions, just ask.

### CODE REVIEW

File: src/main/java/BlueTurtle/summarizers/Summarizer.java [hunk 1]
```diff
@@ -29,6 +32,34 @@ public Summarizer(String packageName) {
...
+		case "PMD":
+			numberOfPMDWarnings++;
+			numberOfWarnings++;
+			break;
+		case "FindBugs":
+			numberOfFindBugsWarnings++;
+			numberOfWarnings++;
+			break;
+		default:
+			numberOfWarnings += 0; // nothing should be added.
```
Comment (boninggong) [inline src/main/java/BlueTurtle/summarizers/Summarizer.java]: What does this do?

File: src/test/java/BlueTurtle/summarizers/ComponentSummarizerTest.java [hunk 2]
```diff
@@ -156,12 +158,36 @@ public void testSummarizingWarningDiffrentPackage() {
 	}
 
 	/**
-	 * Test equal with differnt object.
+	 * Test equal with different object.
```
Comment (boninggong) [inline src/test/java/BlueTurtle/summarizers/ComponentSummarizerTest.java]: Good to see that you are paying attention to spelling mistakes!

File: src/main/java/BlueTurtle/summarizers/Summarizer.java (hunk 1 above)
Comment (ClintonCao) [inline src/main/java/BlueTurtle/summarizers/Summarizer.java]: Read the JavaDoc :stuck_out_tongue:

**Comment by boninggong** [comment]
Goodevening mr Cao,

I see that you updated the analyzer according to the needs of the visualizer, thank you! We now can use these statistics for the visualization. These will make the visualizations even more clear :)

I say this pull request is mergeable!

My sincerely,

Boning Gong

File: src/main/java/BlueTurtle/summarizers/Summarizer.java (hunk 1 above)
Comment (boninggong) [inline src/main/java/BlueTurtle/summarizers/Summarizer.java]: Ohh I see, you needed a default case, oke

Comment (TimBuckers) [inline src/main/java/BlueTurtle/summarizers/Summarizer.java]: The SAT gives a warning when there is no default case. Here the default case should never be triggered, so it's good practice to throw some kind of error if this happens.

Comment (ClintonCao) [inline src/main/java/BlueTurtle/summarizers/Summarizer.java]: Ahh that is right, I will update the code.

**Comment by SunweiWang** [comment]
It's okay to merge it.

**Pull request merged**